Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -859,22 +856,34 @@ internal void OnSerializeAllSafely(bool initialState, NetworkWriter ownerWriter,
             }
         }
 
-        internal ulong GetDirtyMask(bool initialState)
+        internal ulong GetDirtyComponentsMask()
         {
             // loop through all components only once and then write dirty+payload into the writer afterwards
             ulong dirtyComponentsMask = 0L;
             NetworkBehaviour[] components = NetworkBehaviours;
             for (int i = 0; i < components.Length; ++i)
             {
                 NetworkBehaviour comp = components[i];
-                if (initialState || comp.IsDirty())
+                if (comp.IsDirty())
                 {
                     dirtyComponentsMask |= (ulong)(1L << i);
                 }
             }
 
             return dirtyComponentsMask;
         }
+        internal ulong GetIntialComponentsMask()
```

Comment:
is this just for cleaner code, or why do we have two functions now?

Reply:
Yes, now that they are outside of `OnSerializeAllSafely` they don't need to depend on the `bool initialState`. The 2 places that use `OnSerializeAllSafely` call different ComponentsMasks